### ISSUE TITLE
🐛 False positive unused export

### ISSUE THREAD

**Issue by DreierF** [webpro-nl/knip#890]
### Prerequisites

- [X] I'm using the latest version
- [X] I've read the relevant [documentation](https://knip.dev)
- [X] I've searched for [existing issues](https://github.com/webpro-nl/knip/issues?q=is%3Aissue)
- [X] I've checked the [list of known issues](https://knip.dev/reference/known-issues)
- [X] I've read the [issue reproduction guide](https://knip.dev/guides/issue-reproduction)

### Reproduction url

https://stackblitz.com/edit/github-99dfwi73-hxo2rmv8?file=index.ts

### Reproduction access

- [X] I've made sure the reproduction is publicly accessible

### Description of the issue

**Observed behavior**
The example produces:
```
Unused exports (1)
bar  mod1  function  mod1.ts:5:17
```
while `bar` is actually used in `index.ts`.

**Expected behavior**
No error

**Comment by webpro** [comment]
Thanks for the repro. Personally I wouldn't call this "barrel-like", btw.

Either way, it's a bit of an unfortunate case:

- By default, Knip assumes all members of `mod1Exports` used because of the default heuristics: https://knip.dev/guides/namespace-imports
- Then there's `--include nsExports` but the destructured `...mod1Exports` is something Knip is unable to follow further.
- And there's `--include-libs` to hand it over to TypeScript internals (which should be documented better for this use case), but even `LS.findReferences` doesn't find it. You could verify in your IDE with a "Find All References" from `export function bar`: this won't bring you to it's usage.

**Comment by DreierF** [comment]
Yeah, didn't a more fitting term for this construct. However IntelliJ is able to follow along this usage path, but it might use an internal heuristic there.

**Comment by DreierF** [comment]
In case someone runs into the same issue: I found a workaround now that is also understood by knip.

I replaced the `export const AllExports = { ...mod1Exports,  ...mod2Exports };` with an intermediate module (`AllExports.ts`) that re-exports all the symbols.
```
export * from './mod1';
export * from './mod2';
```

and then `export * as` to export those as a namespace.
```
export * as AllExports from './AllExports';
```

**Comment by webpro** [comment]
To be fair, that's not a workaround. That's how ES Modules are designed to be used.

Let's close this one :)
